internal/task/task/bs/add_spdk_target.go

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveAdm
+ * Created Date: 2022-02-08
+ * Author: Jingli Chen (Wine93)
+ */
+
+package bs
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/opencurve/curveadm/cli/cli"
+	comm "github.com/opencurve/curveadm/internal/common"
+	"github.com/opencurve/curveadm/internal/configure"
+	"github.com/opencurve/curveadm/internal/task/scripts"
+	"github.com/opencurve/curveadm/internal/task/step"
+	"github.com/opencurve/curveadm/internal/task/task"
+)
+
+type SpdkTargetOption struct {
+	Host           string
+	User           string
+	BinaryPath     string
+	Cpumask        string
+	Sockname       string
+	ISCSIlogpath   string
+	Create         bool
+	Tid            string
+	Volume         string
+	OriginBdevname string
+	CurveBdevname  string
+	RpcPath        string
+	Size           uint64
+	Blocksize      uint64
+	CbdPath        string
+	IsExcusive     bool
+	TargetIP       string
+	TargetPort     uint64
+	InitiatorIP    string
+	InitiatorMask  uint64
+	VolumeName     string
+}
```

Comment:
Some parameters are not necessary to pass, such as BinaryPath, sockname and ISCSIlogpath etc.. The location can be determined when packaging the image and some special storage locations can be determined in the script.